smithy-kotlin-codegen/build.gradle.kts

```diff
@@ -53,6 +53,7 @@ val generateSdkRuntimeVersion by tasks.registering {
 // unlike the runtime, smithy-kotlin codegen package is not expected to run on Android...we can target 1.8
 tasks.compileKotlin {
     kotlinOptions.jvmTarget = "1.8"
+    kotlinOptions.freeCompilerArgs += "-opt-in=kotlin.RequiresOptIn"
```

Comment:
I'm not sure what you mean. I added this specifically because of the new `codegenReq` function which uses (still-experimental) contracts. There should be no other effect on our codegen execution.